phpmyadmin/content.md

```diff
@@ -0,0 +1,133 @@
+
+```console
+$ docker run --name myadmin -d -e PMA_ARBITRARY=1 -p 8080:80 %%IMAGE%%
+```
+
+## Usage with docker-compose and arbitrary server
+
+This will run phpMyAdmin with arbitrary server - allowing you to specify MySQL/MariaDB server on login page.
+
+%%STACK%%
+
+## Adding Custom Configuration
+
+You can add your own custom config.inc.php settings (such as Configuration Storage setup) by creating a file named "config.user.inc.php" with the various user defined settings in it, and then linking it into the container using `-v /some/local/directory/config.user.inc.php:/etc/phpmyadmin/config.user.inc.php`.
+
+On the `docker run` line like this:
+
+```console
+$ docker run --name myadmin -d --link mysql_db_server:db -p 8080:80 -v /some/local/directory/config.user.inc.php:/etc/phpmyadmin/config.user.inc.php %%IMAGE%%
+```
+
+See the following links for config file information:
+
+-	https://docs.phpmyadmin.net/en/latest/config.html#config
+-	https://docs.phpmyadmin.net/en/latest/setup.html
+
+## Usage behind reverse proxys
+
+Set the variable `PMA_ABSOLUTE_URI` to the fully-qualified path (`https://pma.example.net/`) where the reverse proxy makes phpMyAdmin available.
+
+## Environment variables summary
+
+-	`PMA_ARBITRARY` - when set to 1 connection to the arbitrary server will be allowed
+-	`PMA_HOST` - define address/host name of the MySQL server
+-	`PMA_VERBOSE` - define verbose name of the MySQL server
+-	`PMA_PORT` - define port of the MySQL server
+-	`PMA_HOSTS` - define comma separated list of address/host names of the MySQL servers
+-	`PMA_VERBOSES` - define comma separated list of verbose names of the MySQL servers
+-	`PMA_PORTS` - define comma separated list of ports of the MySQL servers
+-	`PMA_USER` and `PMA_PASSWORD` - define username to use for config authentication method
+-	`PMA_ABSOLUTE_URI` - define user-facing URI
+-	`HIDE_PHP_VERSION` - if defined, will hide the php version (`expose_php = Off`). Set to any value (such as HIDE_PHP_VERSION=true).
+-	`UPLOAD_LIMIT` - if set, will override the default value for apache and php-fpm (default value is 2048 kb)
+-	`PMA_CONFIG_BASE64` - if set, will override the default config.inc.php with the base64 decoded contents of the variable
+-	`PMA_USER_CONFIG_BASE64` - if set, will override the default config.user.inc.php with the base64 decoded contents of the variable
+
+For usage with Docker secrets, appending `_FILE` to the `PMA_PASSWORD` environment variable is allowed (it overrides `PMA_PASSWORD` if it is set):
+
+```console
+$ docker run --name myadmin -d -e PMA_PASSWORD_FILE=/run/secrets/db_password.txt -p 8080:80 phpmyadmin/phpmyadmin
+```
+
+#### Variables that can be read from a file using `_FILE`
+
+-	PMA_PASSWORD
+-	MYSQL_ROOT_PASSWORD
+-	MYSQL_PASSWORD
+-	PMA_HOSTS
+-	PMA_HOST
+
+For more detailed documentation see https://docs.phpmyadmin.net/en/latest/setup.html#installing-using-docker
```

Comment:
Given that this has much more detailed documentation about how to use the image, would it make sense to put less information into this image description so that the same information isn't being maintained in two different places?

The `registry` image/directory is a good example of taking this to the extreme (https://hub.docker.com/_/registry) but I think there's probably a happy medium that could be valuable here.

Reply:
I'd keep it for now, because it gives a nice overview of available environment variables.
@williamdes do you may consider to change this in future?

Reply:
I think official phpMyAdmin documentation should always be linked, right ?

Reply:
True, but we don't need to write things here again that are already covered in the phpMyAdmin docs.

Reply:
Sure

But IMO the ENVs should be documented in the place near the implementation because our docs could diverge
So maybe remove the link ?

Reply:
Why not link here from the official documentation? The official documentation could say something like "There is a Docker image, for simple cases you can use `docker run phpmyadmin/phpmyadmin`, for anything more advanced see the full list of supported options at Docker Hub?

I feel strongly we should only document things like this in only one place, and it makes more sense to me to put it on Docker Hub than docs.phpmyadmin.net.

Reply:
Sure, it doesn't matter where the documentation lives but duplication should be avoided. Anyway, I think this is not a blocker to get https://github.com/docker-library/official-images/pull/7288 done.

Reply:
Let's definitely revisit this post-merge :+1: